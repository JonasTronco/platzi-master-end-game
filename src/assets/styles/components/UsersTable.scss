@import '../main.scss';

.users-search {
  @include flexStaSta;
  margin-bottom: $space_x6;

  .input {
    margin-right: $space_x3;
  }

  .btn {
    margin: 23.67px $space_x3 0;
  }
}

@media screen and (max-width: 425px) {
  .users-search {
    flex-wrap: wrap;

    .input,
    .btn {
      width: 100%;
      margin: 0;
      margin-top: $space_x3;
    }

    .input {
      &__label,
      &__field,
      &__error {
        width: 100%;
      }
    }
  }
}
