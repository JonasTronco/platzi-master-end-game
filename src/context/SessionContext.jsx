import { useState, useEffect, createContext, useMemo, useContext } from 'react';
import Router from 'next/router';

const SessionContext = createContext();
export function SessionProvider(props) {
  const [session, setSession] = useState(null);

  useEffect(() => {
    try {
      setSession(JSON.parse(localStorage.getItem('session')));
    } catch (error) {
      Router.push('/');
    }
  }, []);

  const value = useMemo(
    () => ({
      session,
    }),
    [session],
  );

  return <SessionContext.Provider value={value} {...props} />;
}

export function useSession() {
  const context = useContext(SessionContext);
  if (!context) {
    throw new Error('No Session Context Provider');
  }
  return context;
}
