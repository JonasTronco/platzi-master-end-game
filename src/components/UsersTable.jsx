/* eslint-disable jsx-a11y/control-has-associated-label */
import PropTypes from 'prop-types';

import EditIcon from '../assets/icons/edit.svg';
import DeleteIcon from '../assets/icons/delete.svg';
import ActivateIcon from '../assets/icons/activate.svg';

import '../assets/styles/components/UsersTable.scss';

function UsersTable({ data }) {
  const tableContent = data.map((user) => (
    <tr className="table__body__row">
      <td className="table__body__row__cell">
        <img alt="user profile pic" src={user.imageURL} className="table__body__row__cell__image" />
      </td>
      <td className="table__body__row__cell"><p>{user.username}</p></td>
      <td className="table__body__row__cell"><p>{`${user.firstName} ${user.lastName}`}</p></td>
      <td className="table__body__row__cell"><p>{user.type}</p></td>
      {user.isActive === true
        ? (
          <>
            <td className="table__body__row__cell">
              <EditIcon className="table__body__row__cell__icon" />
            </td>
            <td className="table__body__row__cell">
              <DeleteIcon className="table__body__row__cell__icon" />
            </td>
          </>
        )
        : (
          <td className="table__body__row__cell">
            <ActivateIcon className="table__body__row__cell__icon" />
          </td>
        )}
    </tr>
  ));

  return (
    <main className="table__container">
      <table className="table">
        <thead className="table__head">
          <tr className="table__head__row">
            <th className="table__head__row__cell" />
            <th className="table__head__row__cell"><strong>Username</strong></th>
            <th className="table__head__row__cell"><strong>Full Name</strong></th>
            <th className="table__head__row__cell"><strong>Type</strong></th>
            <th className="table__head__row__cell" />
            <th className="table__head__row__cell" />
          </tr>
        </thead>
        <tbody className="table__body">
          {tableContent}
        </tbody>
      </table>
    </main>
  );
}

UsersTable.propTypes = {
  data: PropTypes.arrayOf,
};

UsersTable.defaultProps = {
  data: [
    { firstName: 'alberto', lastName: 'camarena', username: 'alberto.camarena.1234', imageURL: '/default.png', isActive: true, type: 'Doctor' },
    { firstName: 'luis', lastName: 'flores', username: 'luis.flores.5678', imageURL: '/default.png', type: 'Administrator' },
  ],
};

export default UsersTable;
