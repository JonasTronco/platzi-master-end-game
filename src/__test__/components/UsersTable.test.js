import UsersTable from '../../components/UsersTable';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';
import { SessionProvider } from '../../context/SessionContext';
describe('UserTable Component', () => {
  const obj = {
    session: {
      token: '',
      user: {
        id: '',
        username: '',
        typeOfUser: '',
        isActive: true,
        imageURL: '',
        firstName: '',
        lastName: '',
        defaultPath: ''
      }
    }
  };
  test('Should render UserTable component', () => {
    const component = create(
      <SessionProvider value={obj}>
        <UsersTable />
      </SessionProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('<UserTable />', () => {
    const component = mount(
      <SessionProvider value={obj}>
        <UsersTable />
      </SessionProvider>
    );
    expect(component.length).toEqual(1);
  });
});
