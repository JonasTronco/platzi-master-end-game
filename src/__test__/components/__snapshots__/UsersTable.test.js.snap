// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTable Component Should render UserTable component 1`] = `
Array [
  <form
    action="#"
    className="users-search"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="input"
    >
      <label
        className="input__label"
      >
        Document ID
        <input
          className="input__field"
          disabled={false}
          name="documentID"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
    </div>
    
    <button
      className="btn"
      disabled={false}
      type="submit"
    >
      Search
    </button>
    
  </form>,
  "",
]
`;
